src/Helper/Quote.php

```diff
@@ -186,7 +186,7 @@ public function getValidatedItems()
             }
             $type                         = $this->typeHelper->getType($item);
             $stockData                    = $type->getStockData();
-            $productId                    = $type->getItemId();
+            $productId                    = unserialize($item->getAdditionalData())->getShopgateItemNumber();
```

Comment:
I'd use `\Zend_Json_Encoder::XXX($value)` for things like this

Reply:
switched to \Zend\Serializer\Serializer because \Zend_Json_Encoder / Decoder cannot handle classes.